<launch>

    <arg name="config_path" default = "$(find feature_tracker)/../config/euroc/euroc_config.yaml" />
	  <arg name="vins_path" default = "$(find feature_tracker)/../config/../" />
    
    <node name="feature_tracker" pkg="feature_tracker" type="feature_tracker" output="log">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="vins_folder" type="string" value="$(arg vins_path)" />
    </node>

    <node name="vins_estimator" pkg="vins_estimator" type="vins_estimator" output="screen">
       <param name="config_file" type="string" value="$(arg config_path)" />
       <param name="vins_folder" type="string" value="$(arg vins_path)" />
    </node>

    <node name="pose_graph" pkg="pose_graph" type="pose_graph" output="screen">
        <param name="config_file" type="string" value="$(arg config_path)" />
        <param name="visualization_shift_x" type="int" value="0" />
        <param name="visualization_shift_y" type="int" value="0" />
        <param name="skip_cnt" type="int" value="0" />
        <param name="skip_dis" type="double" value="0" />
    </node>
    <include file="$(find vins_estimator)/launch/vins_rviz.launch" />
    	<arg name="sequence_name" default = "MH_04_difficult" />

    <node name="benchmark_publisher" pkg="benchmark_publisher" type="benchmark_publisher" output="screen">
        <param name="data_name" type="string" value="$(find benchmark_publisher)/config/$(arg sequence_name)/data.csv" />
        <remap from="~estimated_odometry" to="/vins_estimator/odometry" />
    </node>

    <!-- <node pkg="rosbag" type="play" name="play_bag_node" args="/home/erge/桌面/开源工程/VINS-mono/MH_01_easy.bag"/> -->

</launch>
